server/controllers/file.js

```diff
@@ -0,0 +1,258 @@
+const { createWriteStream, unlink } = require('fs');
+const jscrypt = require('jscrypt');
+const { Magic } = require('@magic-sdk/admin');
+const { create } = require("ipfs-http-client");
+const path = require('path')
+
+const User = require('../data-repository/user');
+
+const magic =  new Magic(process.env.MAGIC_SECRET_KEY);
+
+const projectId = process.env.INFURA_PROJECT_ID;
+const projectSecret = process.env.INFURA_PROJECT_SECRET;
+const auth = 'Basic ' + Buffer.from(projectId + ':' + projectSecret).toString('base64');
+
+/*
+    Connection to the INFURA IPFS node
+    Get the secret keys from the INFURA IPFS dashboard
+    Website link: https://infura.io/product/ipfs
+*/
+async function ipfsClient() {
+    const ipfs = await create(
+        {
+            host: "ipfs.infura.io",
+            port: 5001,
+            protocol: "https",
+            headers: {
+                authorization: auth,
+            },
+        }
+    );
+    return ipfs;
+}
+
+async function getFile(cid, encryptedPath) {
+    const ipfs = await ipfsClient();
+    const asyncitr = ipfs.cat(cid);
+
+    //create a file under public folder and write the data to it
+    const file = createWriteStream(encryptedPath);
+    //iterate the async iterator
+    for await (const chunk of asyncitr) {
+        file.write(chunk);
+    }
+}
+
+const addFile = async (fileName, filePath) => {
+    const file = readFileSync(filePath);
+    const ipfs = await ipfsClient();
+    const fileAdded = await ipfs.add({ path: fileName, content: file });
+    return fileAdded;
+}
+
+
+module.exports = class DownloadController{
+    static async secureDownload(req, res){
+        console.log("Secure download called");
+        const cid = req.params.cid;
+        const auth = req.params.auth;
+        const metadata = await magic.users.getMetadataByToken(auth);
+        const magic_id = metadata.issuer;
+    
+        try {
+            const file = await User.findOne({ magic_id: magic_id, files: { $elemMatch: { cid: cid } } }, { encryption_key: 1 }).select({ files: { $elemMatch: { cid: cid } } });
```

Comment:
Error while downloading a file: 
![image](https://user-images.githubusercontent.com/63467479/192140930-a4a6fe34-d25d-46a7-a113-b2cb1ae1c07f.png)

Maybe modify the user.js file in the data-repository.